staging/src/k8s.io/client-go/tools/clientcmd/client_config.go

```diff
@@ -131,6 +131,7 @@ func (config *DirectClientConfig) ClientConfig() (*restclient.Config, error) {
 
 	clientConfig := &restclient.Config{}
 	clientConfig.Host = configClusterInfo.Server
+	clientConfig.DisableCompression = true
```

Comment:
the `restclient.Config` passes through quite a few functions before it is used to configure the transport. Is there a function you think is more appropriate for setting this in the `restclient.Config`?